addons/binding/org.openhab.binding.nuki/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,60 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="nuki"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xmlns:thing="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0"
+        xsi:schemaLocation="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0 http://eclipse.org/smarthome/schemas/thing-description-1.0.0.xsd">
+
+    <!-- Nuki Bridge (Bridge Type) -->
+    <bridge-type id="bridge">
+        <label>Nuki Bridge</label>
+        <description>This bridge represents a Nuki Bridge on your local network. Nuki Smart Locks have to be paired via Bluetooth with it.</description>
+        <config-description>
+            <parameter name="ip" type="text" required="true">
+                <label>IP Address</label>
+                <description>The IP Address of the Nuki Bridge.</description>
+            </parameter>
+            <parameter name="port" type="integer" required="true">
+                <label>Port</label>
+                <description>The Port on which the Nuki Bridge REST API runs.</description>
+                <default>8080</default>
+            </parameter>
+            <parameter name="apiToken" type="text" required="true">
+                <label>API Token</label>
+                <description>The API Token configured via the Nuki App when enabling the API/IFTTT.</description>
+            </parameter>
+        </config-description>
+    </bridge-type>
+
+    <!-- Nuki Smart Lock (Thing Type) -->
+    <thing-type id="smartLock">
+        <supported-bridge-type-refs>
+            <bridge-type-ref id="bridge"/>
+        </supported-bridge-type-refs>
+        <label>Nuki Smart Lock</label>
+        <description>Nuki Smart Lock which is paired via Bluetooth to a Nuki Bridge.</description>
+        <channels>
+            <channel id="smartLockOpenClose" typeId="channelTypeSmartLockOpenClose"/>
+            <channel  id="smartLockUnlatchClose" typeId="channelTypeSmartLockUnlatchClose"/>
+        </channels>
+        <config-description>
+            <parameter name="nukiId" type="text" required="true">
+                <label>Nuki ID</label>
+                <description>The ID of the Nuki Smart Lock.</description>
+            </parameter>
+        </config-description>
+    </thing-type>
+
+    <!-- Channel Type -->
+    <channel-type id="channelTypeSmartLockOpenClose">
+        <item-type>Switch</item-type>
+        <label>Nuki Smart Lock Open/Close</label>
+        <description>Nuki Smart Lock channel to open or close the door.</description>
+    </channel-type>
+
+    <channel-type id="channelTypeSmartLockUnlatchClose">
+        <item-type>Switch</item-type>
+        <label>Nuki Smart Lock Unlatch/Close</label>
+        <description>Nuki Smart Lock channel to unlatch or close the door.</description>
```

Comment:
I would suggest to merge channels 2+3 into one:
A Number channel, which takes the states according to the Nuki Bridge API (they use numbers from 0-255), with state options that provide the mapping to strings (1=locked, 2=unlocking,...). Additionally, the very same channel could take the lock actions as commands. I would actually think that the states 2=unlocking, 4=locking and 7=unlatching would map perfectly on the actions 1, 2 & 3, so we would only need to find some numbers for the lock'n'go actions (4 & 5) - maybe they could be done as -2 and -7 (or 1002 and 1007) to say that it is just a different version of 2 and 7?